dbgd.c

```diff
@@ -211,21 +211,55 @@ static int dbgd_mem_read(Dbg__Request *req, Dbg__Response *res)
     res->address = req->address;
     res->has_address = 1;
 
-    res->size = 1; /* FIXME: add word, dword, qword support */
-    res->has_size = 1;
+    res->data.len  = req->size;
+    res->data.data = malloc(res->data.len);
+    res->has_data = 1;
 
-    res->value = *((uint8_t*)(req->address));
-    res->has_value = 1;
+    unsigned int i = 0;
+    unsigned int s = req->size;
+
+    while(s >= 4) {
```

Comment:
I initially intended `size` to mean access width (1,2,4,8). That's why I had made the `value` field a qword to keep it simple. ... But I do think there is value in allowing us to request `size` number of bytes of memory and send it all back in one shot. I'm not sure if the `while` structure here is the best way to do it. Maybe... for some mysterious reason... I want 512 byte accesses of system memory? I would have to send 512 requests to enforce byte accesses. It might be simpler to do something like: `mem_read(addr, size, count)` where total number of bytes read equals `size*count`.